threatgrid/ctia
# Patch existing entities in `POST /bundle/import`

**Epic** https://github.com/advthreat/iroh/issues/7341
Related https://github.com/advthreat/iroh/issues/8207

Current behavior of bundle import on existing entities is to ignore them. This PR changes behavior to patch in this case, and loosens the schema for bundle import so then partial entities are allowed if patching.



<a name="qa">[§](#qa)</a> QA
============================

This PR adds the ability for `POST /bundle/import` to patch entities. The following steps test if this procedure works.

1. Create an existing entity (for example an Indicator) and note its id.
2. Patch this entity using `POST /bundle/import`. For example, patching the `:producer` field of the indicator:

```
POST /bundle/import?patch-existing=true
{:indicators [{:id "<id-from-step-1>" :producer "PATCHED"}]}
```

3. Verify that the result looks something like this (the `"updated"` part is the most important part):
```
{:results [{:result "updated" :id "<id-from-step-1>"}]}
```

4. Look up the indicator via `GET /ctia/indicator` to verify that the `producer` field was updated.





<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================



```
intern: Patch existing entities in POST /bundle/import
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================
